From 840ea01d953ca0ad7629ea66ca0f50685ca06921 Mon Sep 17 00:00:00 2001
From: Denis Lunev <den@openvz.org>
Date: Mon, 29 Dec 2008 20:34:32 +0300
Subject: [PATCH] NFS: NFS super blocks in different VEs should be different

NFS: NFS super blocks in different VEs should be different

Teach nfs_compare_super to this

Bug #265926

Signed-off-by: Denis V. Lunev <den@openvz.org>
Signed-off-by: Vitaliy Gusev <vgusev@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---
 fs/nfs/super.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/fs/nfs/super.c b/fs/nfs/super.c
index cb4e28a..cf38e22 100644
--- a/fs/nfs/super.c
+++ b/fs/nfs/super.c
@@ -1619,6 +1619,10 @@ static int nfs_compare_super(struct super_block *sb, void *data)
 	struct nfs_server *server = sb_mntdata->server, *old = NFS_SB(sb);
 	int mntflags = sb_mntdata->mntflags;
 
+	if (!ve_accessible_strict(old->client->cl_xprt->owner_env,
+				  get_exec_env()))
+		return 0;
+
 	if (!nfs_compare_super_address(old, server))
 		return 0;
 	/* Note: NFS_MOUNT_UNSHARED == NFS4_MOUNT_UNSHARED */
-- 
1.6.0.6

